packages/core/src/sdk/search/state.ts

```diff
@@ -6,10 +6,15 @@ export const useApplySearchState = () => {
 
   return useCallback(
     (url: URL) => {
+      const currentUrl = new URL(window.location.href)
       const newUrl = `${url.pathname}${url.search}`
-      return url.searchParams.has('fuzzy') && url.searchParams.has('operator')
-        ? router.replace(newUrl)
-        : router.push(newUrl)
+      if (currentUrl.href !== newUrl) {
+        if (url.searchParams.has('fuzzy') && url.searchParams.has('operator')) {
+          window.history.replaceState({}, '', newUrl)
+        } else {
+          router.push(newUrl)
+        }
+      }
```

Comment:
suggestion: it apparently is an issue with the Next's `replace` function, right? To simplify the code what do you think about the following suggestion:
```suggestion
      return url.searchParams.has('fuzzy') && url.searchParams.has('operator')
        ? window.history.replaceState(window.history.state, '', newUrl)
        : router.push(newUrl)
```
I've also added the previous state since we only want to change the URL

Reply:
MSTM! thanks for the input here Lari ❤️